sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -1075,4 +1075,113 @@ class CastSuite extends SparkFunSuite with ExpressionEvalHelper {
       checkEvaluation(cast("badvalue", dataType), null)
     }
   }
+
+  private def testIntMaxAndMin(dt: DataType): Unit = {
+    Seq(Int.MaxValue + 1L, Int.MinValue - 1L).foreach { value =>
+      checkExceptionInExpression[ArithmeticException](cast(value, dt), "overflow")
+      checkExceptionInExpression[ArithmeticException](cast(Decimal(value.toString), dt), "overflow")
+      checkExceptionInExpression[ArithmeticException](
+        cast(Literal(value * MICROS_PER_SECOND, TimestampType), dt), "overflow")
+      checkExceptionInExpression[ArithmeticException](
+        cast(Literal(value * 1.5f, FloatType), dt), "overflow")
+      checkExceptionInExpression[ArithmeticException](
+        cast(Literal(value * 1.0, DoubleType), dt), "overflow")
+    }
+  }
+
+  private def testLongMaxAndMin(dt: DataType): Unit = {
+    Seq(Decimal(Long.MaxValue) + Decimal(1), Decimal(Long.MinValue) - Decimal(1)).foreach { value =>
+      checkExceptionInExpression[ArithmeticException](
+        cast(value, dt), "overflow")
+      checkExceptionInExpression[ArithmeticException](
+        cast((value * Decimal(1.1)).toFloat, dt), "overflow")
+      checkExceptionInExpression[ArithmeticException](
+        cast((value * Decimal(1.1)).toDouble, dt), "overflow")
+    }
+  }
+
+  test("Cast to byte with option FAIL_ON_INTEGER_OVERFLOW enabled") {
+    withSQLConf(SQLConf.FAIL_ON_INTEGER_OVERFLOW.key -> "true") {
+      testIntMaxAndMin(ByteType)
```

Comment:
why we need to test `cast int.max +1 to byte`? I think it's good enough to test `cast byte.max +1 to byte`

Reply:
I think it is always good to have more test cases if it doesn't increase the testing time.
For example, 
If casting double to byte is implemented as:
```
val x = doubleValue.toShort
if (x.toShort == x) {
  x.toShort
} else {
  throw new ...
}
```
We can find that it is wrong with this test case, because
```
 (Int.MaxValue+1.0).toShort.toByte == (Int.MaxValue+1.0).toShort
```
is true.